main.cpp

```diff
@@ -1 +1,215 @@
 #include <iostream>
+#include <ctime>
+#include <vector>
+
+/*
+ * Efficient way to check if a player won a tic tac toe game.
+ * The input is a 3x3 array of bool. NULL is the default state
+ * and true/false are for the players.
+ */
+const int PLAYMODE = 0;
+const int SIZE = 3;
+int8_t board[SIZE][SIZE] =
+        {
+                {0,0,-1},
+                {0,-1,0},
+                {-1,0,0},
+        };
+/*
+ * Affiche le tableau de jeu dans la console
+ */
+void PrintArray()
+{
+    std::cout << "---------------" << std::endl;
+    for(auto & i : board)
+    {
+        std::cout << "|";
+        for(signed char j : i)
+        {
+            if(j == -1) std::cout << " X ";
+            else if(j == 1) std::cout << " O ";
+            else std::cout << "   ";
+            std::cout << "|";
+        }
+        std::cout << std::endl;
+    }
+    std::cout << "---------------" << std::endl;
+}
+
+/*
+ * Remplis le tabelau de jeu avec des valeurs aléatoires de -1, 0 ou 1
+ */
+void GenerateRandomArray()
+{
+    srand(time(0));
+    for(auto & i : board)
+    {
+        for(signed char & j : i)
+        {
+            j = (rand()%3) - 1;
+        }
+    }
+}
+
+/*
+ * Initialise le tableau en remplissant toutes ses cases avec un 0
+ */
+void InitializeArray()
+{
+    for(auto & i : board)
+    {
+        for(signed char & j : i)
+        {
+            j = 0;
+        }
+    }
+}
+
+/*
+ * Gère le tour de jeu pour un joueur humain
+ */
+void HumanPlay(bool isCrossPlaying)
+{
+    std::string cell;
+    uint8_t xCoord, yCoord;
+    bool stayInLoop = true;
+
+    isCrossPlaying ? std::cout << "Cross is playing" << std::endl : std::cout << "Circle is playing" << std::endl;
+
+    do {
+        std::cin >> cell;
+        xCoord = cell[0] - '0';
+        yCoord = cell[1] - '0';
+        if(cell.length() == 2)
+        {
+            if(xCoord <= SIZE && xCoord > 0 && yCoord <= SIZE && yCoord > 0)
```

Comment:
Lucky strike here: `xCoord` is unsigned, so if you enter a `.`, it will be `-1`, which is `255`, so it works. But not very nice...